Xamarin.Forms.Platform.iOS/Material/MaterialFrameRenderer.cs

```diff
@@ -1,165 +1,235 @@
-				_packager = null;
-
-				_tracker.NativeControlUpdated -= OnNativeControlUpdated;
-				_tracker.Dispose();
-				_tracker = null;
-			}
-
-			base.Dispose(disposing);
+			ElementChanged?.Invoke(this, e);
 		}
+
 		void OnNativeControlUpdated(object sender, EventArgs eventArgs)
 		{
-
 		}
 
-		void SetupLayer()
+		void UpdateShadow()
 		{
-			float cornerRadius = FrameElement.CornerRadius;
+			// set the default elevation on the first time
+			if (_defaultElevation < 0)
+				_defaultElevation = (nfloat)GetShadowElevation(UIControlState.Normal);
 
-			if (cornerRadius == -1f)
-				cornerRadius = 5f; // default corner radius
+			if (Element.HasShadow)
+				SetShadowElevation(_defaultElevation, UIControlState.Normal);
+			else
+				SetShadowElevation(0, UIControlState.Normal);
+		}
 
-			CornerRadius = cornerRadius;
+		void UpdateCornerRadius()
+		{
+			// set the default radius on the first time
+			if (_defaultCornerRadius < 0)
+				_defaultCornerRadius = CornerRadius;
 
-			if (Element.BackgroundColor == Color.Default)
-				BackgroundColor = UIColor.White;
+			var cornerRadius = Element.CornerRadius;
+			if (cornerRadius < 0)
+				CornerRadius = _defaultCornerRadius;
 			else
-				BackgroundColor = Element.BackgroundColor.ToUIColor();
+				CornerRadius = cornerRadius;
+		}
 
-			if (FrameElement.HasShadow)
-			{
-				Layer.ShadowRadius = 5;
-				SetShadowColor(UIColor.Black, UIControlState.Normal);
-				
-				SetShadowElevation(3, UIControlState.Normal);				
-				//Layer.ShadowColor = UIColor.Black.CGColor;
-			}
-			else
+		void UpdateBorderColor()
+		{
+			// set the default stroke properties on the first time
+			if (_defaultStrokeColor == null)
+				_defaultStrokeColor = GetBorderColorForState(UIControlState.Normal);
+			if (_defaultStrokeWidth < 0)
+				_defaultStrokeWidth = GetBorderWidth(UIControlState.Normal);
+
+			var borderColor = Element.BorderColor;
+			if (borderColor.IsDefault)
 			{
-				//SetShadowColor(UIColor.Black, UIControlState.Normal);
-				SetShadowColor(UIColor.Clear, UIControlState.Normal);
+				SetBorderColor(_defaultStrokeColor, UIControlState.Normal);
+				SetBorderWidth(_defaultStrokeWidth, UIControlState.Normal);
 			}
-			
-
-			if (FrameElement.BorderColor == Color.Default)
-				SetBorderColor(UIColor.Clear, UIControlState.Normal);
 			else
 			{
-				SetBorderColor(FrameElement.BorderColor.ToUIColor(), UIControlState.Normal);
-				SetBorderWidth(3, UIControlState.Normal);
+				SetBorderColor(borderColor.ToUIColor(), UIControlState.Normal);
+				SetBorderWidth(1f, UIControlState.Normal);
 			}
 		}
 
-		public SizeRequest GetDesiredSize(double widthConstraint, double heightConstraint)
+		void UpdateBackgroundColor()
 		{
-			return NativeView.GetSizeRequest(widthConstraint, heightConstraint, 44, 44);
+			if (_defaultBackgroundColor == null)
+				_defaultBackgroundColor = BackgroundColor;
+
+			var bgColor = Element.BackgroundColor;
+			if (bgColor.IsDefault)
+				BackgroundColor = _defaultBackgroundColor;
```

Comment:
Is this the right thing to set? Or do we need to set it on the scheme and then reapply the scheme?